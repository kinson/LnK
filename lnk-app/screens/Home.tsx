import React, { ReactElement, useState } from 'react';
import Head from 'next/head';

import { createShortLink } from '../lib/createShortLink';
import Icon from '../components/Icon';
import LoadingIcon from '../components/LoadingIcon';
import UrlInput from '../components/UrlInput';
import ShortLink from '../components/ShortLink';
import ShortenButton from '../components/ShortenButton';
import CopyButton from '../components/CopyButton';
import Alert from '../components/Alert';
import DarkModeToggle from '../components/DarkModeToggle';
import { Theme, ThemeContext } from '../contexts/ThemeContext';

interface Props {
  redirectError?: boolean;
}

export default function HomeScreen({ redirectError }: Props): ReactElement {
  const [url, setUrl] = useState<string | null>(null);
  const [error, setError] = useState<string | null>(null);
  const [loading, setLoading] = useState<boolean>(false);
  const [shortLink, setShortLink] = useState<string | null>(null);

  const shorten = async () => {
    if (url === null) return;

    setLoading(true);
    setError(null);

    try {
      const link = await createShortLink(url);
      setShortLink(link);
    } catch (err) {
      setError(err.message);
    } finally {
      setLoading(false);
    }
  };

  const resetForm = () => {
    setUrl(null);
    setError(null);
    setShortLink(null);
  };

  const updateUrl = (event: React.FormEvent<HTMLInputElement>) => {
    setUrl(event.currentTarget.value);
  };

  const labelText = shortLink ? 'Your New Shortened Url' : 'Your Long Url';

  return (
    <div>
      <Head>
        <meta charSet="utf-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />

        <title>LnK</title>

        <meta name="description" content="Yet another URL shortener" />
        <meta name="author" content="Sam Hunter" />
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <ThemeContext.Consumer>
        {({ theme }) => (
          <div className={theme === Theme.DARK ? 'dark' : ''}>
            <main className="w-full mx-auto h-screen transition-colors dark:bg-gray-700">
              <div className="w-11/12 md:w-8/12 lg:w-5/12 mx-auto pt-8 md:pt-16 lg:pt-36">
                {redirectError && (
                  <Alert>Could not find the link you are looking for.</Alert>
                )}

                <div className="flex flex-row items-center">
                  <div className="mr-2 dark:text-gray-200">
                    <Icon size="large" />
                  </div>

                  <h1 className="text-4xl md:text-5xl lg:text-6xl text-center dark:text-gray-200 ml-2">
                    LnK
                  </h1>

                  <DarkModeToggle />
                </div>
                <h2 className="text-lg dark:text-gray-200 my-8">
                  Got a long link? Just paste it below and click shorten âœ¨
                </h2>

                <div className="w-full bg-darkGray bg-opacity-30 dark:bg-opacity-40 shadow-md p-8 rounded-md">
                  <form className="flex flex-col">
                    <div className="py-1 flex items-center justify-between">
                      <label
                        htmlFor="url-input"
                        className="text-darkPurple dark:text-gray-200 opacity-70"
                      >
                        {labelText}
                      </label>
                      <p className="text-red-600">{error ?? ''}</p>
                    </div>

                    <div className="h-20 flex items-center justify-center w-full flex-col">
                      <UrlInput
                        url={url}
                        updateUrl={updateUrl}
                        hidden={!!(loading || shortLink)}
                        onEnterPress={shorten}
                      />

                      <ShortLink hidden={!shortLink} link={shortLink} />

                      <div className="mx-auto">
                        <LoadingIcon show={loading} />
                      </div>
                    </div>

                    <div className="flex flex-row w-max mx-auto mt-8">
                      <ShortenButton
                        disabled={!!(shortLink || loading || !url)}
                        onClick={shorten}
                      />
                      <CopyButton
                        disabled={!!(loading || !shortLink)}
                        link={shortLink}
                      />
                    </div>
                  </form>

                  {!!shortLink && (
                    <div className="mx-auto w-max mt-10">
                      <button
                        type="button"
                        className="underline text-darkPurple dark:text-gray-300"
                        onClick={resetForm}
                      >
                        Shorten Another Link
                      </button>
                    </div>
                  )}
                </div>
              </div>
            </main>
          </div>
        )}
      </ThemeContext.Consumer>
    </div>
  );
}
