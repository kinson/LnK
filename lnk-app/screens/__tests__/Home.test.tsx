import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../Home';

describe('Home screen tests', () => {
  it('should render the Home screen', () => {
    const component = renderer.create(<Home />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should display an error message when the long url cannot be found', () => {
    const component = renderer.create(<Home redirectError />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
