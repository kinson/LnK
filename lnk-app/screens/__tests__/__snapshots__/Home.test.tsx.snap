// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen tests should display an error message when the long url cannot be found 1`] = `
<div>
  <div
    className=""
  >
    <main
      className="w-full mx-auto h-screen transition-colors dark:bg-gray-700 false"
    >
      <div
        className="w-11/12 md:w-8/12 lg:w-5/12 mx-auto pt-8 md:pt-16 lg:pt-36"
      >
        <p
          className="my-8 rounded-md bg-red-200 border-2 border-red-400 text-red-700 p-4"
        >
          Could not find the link you are looking for.
        </p>
        <div
          className="flex flex-row items-center"
        >
          <div
            className="mr-2 dark:text-white"
          >
            <svg
              className="max-h-16"
              version="1.1"
              viewBox="0 0 202 211"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink"
            >
              <g
                fill="none"
                stroke="currentColor"
                strokeWidth="8"
                transform="translate(6.000000, 6.000000)"
              >
                <path
                  d="M44.1456633,53.6451202 L44.0114514,48.4191476 C43.8554831,42.3460178 48.6522902,37.2963361 54.72542,37.1403677 C54.8195359,37.1379507 54.9136788,37.136742 55.0078257,37.136742 L174.7217,37.136742 C180.796832,37.136742 185.7217,42.0616098 185.7217,48.136742 L185.7217,90.136742 C185.7217,96.2118743 180.796832,101.136742 174.7217,101.136742 L54.4928782,101.136742 C48.507847,101.136742 43.6211034,96.3516523 43.4953082,90.3679432 L43.0445969,68.9289161 L43.0445969,68.9289161"
                  id="Rectangle"
                  transform="translate(114.383148, 69.136742) rotate(-38.000000) translate(-114.383148, -69.136742) "
                />
                <path
                  d="M5.1400357,113.858202 L5.07884581,108.966168 C5.0028636,102.891511 9.86575039,97.9054323 15.9404075,97.8294501 C15.9862645,97.8288765 16.0321248,97.8285897 16.0779854,97.8285897 L135.939536,97.8285897 C142.014668,97.8285897 146.939536,102.753457 146.939536,108.82859 L146.939536,127.579032 L146.939536,127.579032 L146.939536,150.82859 C146.939536,156.903722 142.014668,161.82859 135.939536,161.82859 L15.829257,161.82859 C9.79736303,161.82859 4.89059053,156.971009 4.82981533,150.939422 L4.61559465,129.679257 L4.61559465,129.679257"
                  id="Rectangle"
                  transform="translate(75.777565, 129.828590) rotate(142.000000) translate(-75.777565, -129.828590) "
                />
              </g>
            </svg>
          </div>
          <h1
            className="text-6xl text-center dark:text-white ml-2"
          >
            LnK
          </h1>
          <button
            className="ml-auto mr-2 flex items-center cursor-pointer"
            id="dark-mode-toggle-container"
            onClick={[Function]}
            type="button"
          >
            <div
              className="relative"
            >
              <input
                aria-label="Dark Mode Toggle"
                className="sr-only"
                defaultChecked={false}
                id="dark-mode-toggle"
                type="checkbox"
              />
              <div
                className="block bg-gray-600 w-14 h-8 rounded-full"
              />
              <div
                className="false absolute left-1 top-1 bg-white dark:bg-purple w-6 h-6 rounded-full flex items-center justify-center transition-all"
              >
                <svg
                  className="h-4 text-black"
                  fill="currentColor"
                  id="sun-icon"
                  stroke="currentColor"
                  version="1.1"
                  viewBox="0 0 170 170"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    id="Group-2"
                    transform="translate(80.000000, 79.000000) rotate(-4.000000) translate(-80.000000, -79.000000) translate(1.000000, -0.000000)"
                  >
                    <circle
                      cx="79"
                      cy="79"
                      id="Oval"
                      r="22.5"
                    />
                    <g
                      id="Group"
                      stroke="currentColor"
                      strokeLinecap="square"
                      strokeWidth="5"
                    >
                      <line
                        x1="111.085701"
                        x2="139.247251"
                        y1="51.5309469"
                        y2="27.5243795"
                      />
                      <line
                        x1="121.11157"
                        x2="158"
                        y1="82.2644628"
                        y2="85.2024793"
                      />
                      <line
                        transform="translate(118.472337, 125.166476) rotate(90.000000) translate(-118.472337, -125.166476) "
                        x1="104.160729"
                        x2="132.783944"
                        y1="137.400592"
                        y2="112.93236"
                      />
                      <line
                        x1="75.7355372"
                        x2="72.7975207"
                        y1="121.11157"
                        y2="158"
                      />
                      <line
                        transform="translate(32.833524, 118.472337) rotate(180.000000) translate(-32.833524, -118.472337) "
                        x1="18.7527491"
                        x2="46.9142993"
                        y1="130.475621"
                        y2="106.469053"
                      />
                      <line
                        x1="36.8884298"
                        x2="0"
                        y1="75.7355372"
                        y2="72.7975207"
                      />
                      <line
                        transform="translate(39.527663, 32.833524) rotate(270.000000) translate(-39.527663, -32.833524) "
                        x1="25.2160557"
                        x2="53.8392707"
                        y1="45.0676403"
                        y2="20.5994081"
                      />
                      <line
                        x1="82.2644628"
                        x2="85.2024793"
                        y1="36.8884298"
                        y2="0"
                      />
                    </g>
                  </g>
                </svg>
              </div>
            </div>
          </button>
        </div>
        <h2
          className="text-lg dark:text-white my-8"
        >
          Got a long link? We'll take care of that.
        </h2>
        <div
          className="w-full bg-darkGray bg-opacity-20 shadow-md p-8 rounded-md"
        >
          <form
            className="flex flex-col"
          >
            <div
              className="py-1 flex items-center justify-between"
            >
              <label
                className="text-darkPurple dark:text-gray-200 opacity-70"
                htmlFor="url-input"
              >
                Your Long Url
              </label>
              <p
                className="text-red-600"
              >
                
              </p>
            </div>
            <div
              className="h-20 flex items-center justify-center w-full flex-col"
            >
              <input
                className="placeholder-gray-300 text-darkPurple dark:text-white text-md relative bg-white dark:bg-gray-600 rounded border-0 shadow outline-none focus:outline-none w-full transition-opacity duration-300 p-3"
                id="url-input"
                name="url-input"
                onChange={[Function]}
                placeholder="https://google.com/?q=A+Long+Query+You+Want+To+Share+With..."
                type="text"
                value=""
              />
              <a
                className="text-violet dark:text-lavender text-md relative bg-white dark:bg-gray-600 xrounded border-0 shadow outline-none focus:outline-none w-full transition-opacity duration-300 h-0 p-0 opacity-0"
                href="#"
              />
              <div
                className="mx-auto"
              >
                <div
                  className="rounded-full w-14 bg-purple flex items-center justify-center transition-opacity duration-300 delay-75 opacity-0 h-0"
                >
                  <div
                    className="animate-pulse text-white"
                  >
                    <svg
                      className="max-h-8"
                      version="1.1"
                      viewBox="0 0 202 211"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlnsXlink="http://www.w3.org/1999/xlink"
                    >
                      <g
                        fill="none"
                        stroke="currentColor"
                        strokeWidth="8"
                        transform="translate(6.000000, 6.000000)"
                      >
                        <path
                          d="M44.1456633,53.6451202 L44.0114514,48.4191476 C43.8554831,42.3460178 48.6522902,37.2963361 54.72542,37.1403677 C54.8195359,37.1379507 54.9136788,37.136742 55.0078257,37.136742 L174.7217,37.136742 C180.796832,37.136742 185.7217,42.0616098 185.7217,48.136742 L185.7217,90.136742 C185.7217,96.2118743 180.796832,101.136742 174.7217,101.136742 L54.4928782,101.136742 C48.507847,101.136742 43.6211034,96.3516523 43.4953082,90.3679432 L43.0445969,68.9289161 L43.0445969,68.9289161"
                          id="Rectangle"
                          transform="translate(114.383148, 69.136742) rotate(-38.000000) translate(-114.383148, -69.136742) "
                        />
                        <path
                          d="M5.1400357,113.858202 L5.07884581,108.966168 C5.0028636,102.891511 9.86575039,97.9054323 15.9404075,97.8294501 C15.9862645,97.8288765 16.0321248,97.8285897 16.0779854,97.8285897 L135.939536,97.8285897 C142.014668,97.8285897 146.939536,102.753457 146.939536,108.82859 L146.939536,127.579032 L146.939536,127.579032 L146.939536,150.82859 C146.939536,156.903722 142.014668,161.82859 135.939536,161.82859 L15.829257,161.82859 C9.79736303,161.82859 4.89059053,156.971009 4.82981533,150.939422 L4.61559465,129.679257 L4.61559465,129.679257"
                          id="Rectangle"
                          transform="translate(75.777565, 129.828590) rotate(142.000000) translate(-75.777565, -129.828590) "
                        />
                      </g>
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="flex flex-row w-max mx-auto mt-8"
            >
              <button
                className="bg-purple mr-0.5 px-3 py-2 rounded-tl-md rounded-bl-md text-white bg-opacity-60 text-opacity-70"
                disabled={true}
                onClick={[Function]}
                type="button"
              >
                Shorten
              </button>
              <button
                className="bg-violet ml-0.5 px-3 py-2 rounded-tr-md rounded-br-md text-white bg-opacity-60 text-opacity-70"
                disabled={true}
                onClick={[Function]}
                type="button"
              >
                & Copy
              </button>
              <input
                className="h-0 w-0 focus:outline-none"
                readOnly={true}
                value=""
              />
            </div>
          </form>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`Home screen tests should render the Home screen 1`] = `
<div>
  <div
    className=""
  >
    <main
      className="w-full mx-auto h-screen transition-colors dark:bg-gray-700 false"
    >
      <div
        className="w-11/12 md:w-8/12 lg:w-5/12 mx-auto pt-8 md:pt-16 lg:pt-36"
      >
        <div
          className="flex flex-row items-center"
        >
          <div
            className="mr-2 dark:text-white"
          >
            <svg
              className="max-h-16"
              version="1.1"
              viewBox="0 0 202 211"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink"
            >
              <g
                fill="none"
                stroke="currentColor"
                strokeWidth="8"
                transform="translate(6.000000, 6.000000)"
              >
                <path
                  d="M44.1456633,53.6451202 L44.0114514,48.4191476 C43.8554831,42.3460178 48.6522902,37.2963361 54.72542,37.1403677 C54.8195359,37.1379507 54.9136788,37.136742 55.0078257,37.136742 L174.7217,37.136742 C180.796832,37.136742 185.7217,42.0616098 185.7217,48.136742 L185.7217,90.136742 C185.7217,96.2118743 180.796832,101.136742 174.7217,101.136742 L54.4928782,101.136742 C48.507847,101.136742 43.6211034,96.3516523 43.4953082,90.3679432 L43.0445969,68.9289161 L43.0445969,68.9289161"
                  id="Rectangle"
                  transform="translate(114.383148, 69.136742) rotate(-38.000000) translate(-114.383148, -69.136742) "
                />
                <path
                  d="M5.1400357,113.858202 L5.07884581,108.966168 C5.0028636,102.891511 9.86575039,97.9054323 15.9404075,97.8294501 C15.9862645,97.8288765 16.0321248,97.8285897 16.0779854,97.8285897 L135.939536,97.8285897 C142.014668,97.8285897 146.939536,102.753457 146.939536,108.82859 L146.939536,127.579032 L146.939536,127.579032 L146.939536,150.82859 C146.939536,156.903722 142.014668,161.82859 135.939536,161.82859 L15.829257,161.82859 C9.79736303,161.82859 4.89059053,156.971009 4.82981533,150.939422 L4.61559465,129.679257 L4.61559465,129.679257"
                  id="Rectangle"
                  transform="translate(75.777565, 129.828590) rotate(142.000000) translate(-75.777565, -129.828590) "
                />
              </g>
            </svg>
          </div>
          <h1
            className="text-6xl text-center dark:text-white ml-2"
          >
            LnK
          </h1>
          <button
            className="ml-auto mr-2 flex items-center cursor-pointer"
            id="dark-mode-toggle-container"
            onClick={[Function]}
            type="button"
          >
            <div
              className="relative"
            >
              <input
                aria-label="Dark Mode Toggle"
                className="sr-only"
                defaultChecked={false}
                id="dark-mode-toggle"
                type="checkbox"
              />
              <div
                className="block bg-gray-600 w-14 h-8 rounded-full"
              />
              <div
                className="false absolute left-1 top-1 bg-white dark:bg-purple w-6 h-6 rounded-full flex items-center justify-center transition-all"
              >
                <svg
                  className="h-4 text-black"
                  fill="currentColor"
                  id="sun-icon"
                  stroke="currentColor"
                  version="1.1"
                  viewBox="0 0 170 170"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    id="Group-2"
                    transform="translate(80.000000, 79.000000) rotate(-4.000000) translate(-80.000000, -79.000000) translate(1.000000, -0.000000)"
                  >
                    <circle
                      cx="79"
                      cy="79"
                      id="Oval"
                      r="22.5"
                    />
                    <g
                      id="Group"
                      stroke="currentColor"
                      strokeLinecap="square"
                      strokeWidth="5"
                    >
                      <line
                        x1="111.085701"
                        x2="139.247251"
                        y1="51.5309469"
                        y2="27.5243795"
                      />
                      <line
                        x1="121.11157"
                        x2="158"
                        y1="82.2644628"
                        y2="85.2024793"
                      />
                      <line
                        transform="translate(118.472337, 125.166476) rotate(90.000000) translate(-118.472337, -125.166476) "
                        x1="104.160729"
                        x2="132.783944"
                        y1="137.400592"
                        y2="112.93236"
                      />
                      <line
                        x1="75.7355372"
                        x2="72.7975207"
                        y1="121.11157"
                        y2="158"
                      />
                      <line
                        transform="translate(32.833524, 118.472337) rotate(180.000000) translate(-32.833524, -118.472337) "
                        x1="18.7527491"
                        x2="46.9142993"
                        y1="130.475621"
                        y2="106.469053"
                      />
                      <line
                        x1="36.8884298"
                        x2="0"
                        y1="75.7355372"
                        y2="72.7975207"
                      />
                      <line
                        transform="translate(39.527663, 32.833524) rotate(270.000000) translate(-39.527663, -32.833524) "
                        x1="25.2160557"
                        x2="53.8392707"
                        y1="45.0676403"
                        y2="20.5994081"
                      />
                      <line
                        x1="82.2644628"
                        x2="85.2024793"
                        y1="36.8884298"
                        y2="0"
                      />
                    </g>
                  </g>
                </svg>
              </div>
            </div>
          </button>
        </div>
        <h2
          className="text-lg dark:text-white my-8"
        >
          Got a long link? We'll take care of that.
        </h2>
        <div
          className="w-full bg-darkGray bg-opacity-20 shadow-md p-8 rounded-md"
        >
          <form
            className="flex flex-col"
          >
            <div
              className="py-1 flex items-center justify-between"
            >
              <label
                className="text-darkPurple dark:text-gray-200 opacity-70"
                htmlFor="url-input"
              >
                Your Long Url
              </label>
              <p
                className="text-red-600"
              >
                
              </p>
            </div>
            <div
              className="h-20 flex items-center justify-center w-full flex-col"
            >
              <input
                className="placeholder-gray-300 text-darkPurple dark:text-white text-md relative bg-white dark:bg-gray-600 rounded border-0 shadow outline-none focus:outline-none w-full transition-opacity duration-300 p-3"
                id="url-input"
                name="url-input"
                onChange={[Function]}
                placeholder="https://google.com/?q=A+Long+Query+You+Want+To+Share+With..."
                type="text"
                value=""
              />
              <a
                className="text-violet dark:text-lavender text-md relative bg-white dark:bg-gray-600 xrounded border-0 shadow outline-none focus:outline-none w-full transition-opacity duration-300 h-0 p-0 opacity-0"
                href="#"
              />
              <div
                className="mx-auto"
              >
                <div
                  className="rounded-full w-14 bg-purple flex items-center justify-center transition-opacity duration-300 delay-75 opacity-0 h-0"
                >
                  <div
                    className="animate-pulse text-white"
                  >
                    <svg
                      className="max-h-8"
                      version="1.1"
                      viewBox="0 0 202 211"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlnsXlink="http://www.w3.org/1999/xlink"
                    >
                      <g
                        fill="none"
                        stroke="currentColor"
                        strokeWidth="8"
                        transform="translate(6.000000, 6.000000)"
                      >
                        <path
                          d="M44.1456633,53.6451202 L44.0114514,48.4191476 C43.8554831,42.3460178 48.6522902,37.2963361 54.72542,37.1403677 C54.8195359,37.1379507 54.9136788,37.136742 55.0078257,37.136742 L174.7217,37.136742 C180.796832,37.136742 185.7217,42.0616098 185.7217,48.136742 L185.7217,90.136742 C185.7217,96.2118743 180.796832,101.136742 174.7217,101.136742 L54.4928782,101.136742 C48.507847,101.136742 43.6211034,96.3516523 43.4953082,90.3679432 L43.0445969,68.9289161 L43.0445969,68.9289161"
                          id="Rectangle"
                          transform="translate(114.383148, 69.136742) rotate(-38.000000) translate(-114.383148, -69.136742) "
                        />
                        <path
                          d="M5.1400357,113.858202 L5.07884581,108.966168 C5.0028636,102.891511 9.86575039,97.9054323 15.9404075,97.8294501 C15.9862645,97.8288765 16.0321248,97.8285897 16.0779854,97.8285897 L135.939536,97.8285897 C142.014668,97.8285897 146.939536,102.753457 146.939536,108.82859 L146.939536,127.579032 L146.939536,127.579032 L146.939536,150.82859 C146.939536,156.903722 142.014668,161.82859 135.939536,161.82859 L15.829257,161.82859 C9.79736303,161.82859 4.89059053,156.971009 4.82981533,150.939422 L4.61559465,129.679257 L4.61559465,129.679257"
                          id="Rectangle"
                          transform="translate(75.777565, 129.828590) rotate(142.000000) translate(-75.777565, -129.828590) "
                        />
                      </g>
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="flex flex-row w-max mx-auto mt-8"
            >
              <button
                className="bg-purple mr-0.5 px-3 py-2 rounded-tl-md rounded-bl-md text-white bg-opacity-60 text-opacity-70"
                disabled={true}
                onClick={[Function]}
                type="button"
              >
                Shorten
              </button>
              <button
                className="bg-violet ml-0.5 px-3 py-2 rounded-tr-md rounded-br-md text-white bg-opacity-60 text-opacity-70"
                disabled={true}
                onClick={[Function]}
                type="button"
              >
                & Copy
              </button>
              <input
                className="h-0 w-0 focus:outline-none"
                readOnly={true}
                value=""
              />
            </div>
          </form>
        </div>
      </div>
    </main>
  </div>
</div>
`;
